### PR TITLE
Db restructure

### PR DESCRIPTION
**Pull request by scholtz-gnome**

### CODE REVIEW

File: db/migrations/20210517100921_create_community_table.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import { Knex } from "knex";
+
+export async function up(knex: Knex): Promise<void> {
+  return knex.schema.createTable("community", (table) => {
+    table.increments().primary();
+    table.string("name").notNullable();
+    table.string("about").defaultTo(null);
```
Comment (jcqvisser) [inline db/migrations/20210517100921_create_community_table.ts:7]: Consider using `table.text("about")`. Else the column is of type `VARCHAR(255)`, which can only hold 255 characters...

Comment (scholtz-gnome) [reply db/migrations/20210517100921_create_community_table.ts:7]: Thanks, will do.

File: db/migrations/20210517101603_create_song_file_table.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { Knex } from "knex";
+
+export async function up(knex: Knex): Promise<void> {
+  return knex.schema.createTable("song_file", (table) => {
```
Comment (jcqvisser) [inline db/migrations/20210517101603_create_song_file_table.ts:4]: Can a file belong to multiple songs? If not, consider dropping this table in favour of having a `song_id` column on the `file` table.

File: db/migrations/20210517102049_create_user_community_table.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+export async function up(knex: Knex): Promise<void> {
+  return knex.schema.createTable("user_community", (table) => {
+    table.increments().primary();
+    table.integer("user_id").references("id").inTable("user").notNullable();
+    table
+      .integer("community_id")
+      .references("id")
+      .inTable("community")
+      .notNullable();
```
Comment (jcqvisser) [inline db/migrations/20210517102049_create_user_community_table.ts:11]: Consider adding `.index()` for `community_id` and `user_id`.

Adding indexes on `*_id` columns of linking tables is quite valuable - it speeds up queries involving that table significantly.

**Review by jcqvisser — CHANGES_REQUESTED** [review]
Tables look good. Consider suggestions, though.
My main issue is the test suite not passing - it's good practise to keep that green.

**Pull request merged**